plugins/ltac2/dune

```diff
@@ -3,6 +3,6 @@
  (public_name coq-core.plugins.ltac2)
  (synopsis "Ltac2 plugin")
  (modules_without_implementation tac2expr tac2qexpr tac2types)
- (libraries vernac))
+ (libraries vernac coq-core.plugins.cc))
```

Comment:
ltac2_plugin is not supposed to depend on ltac1 but cc depends on ltac1
Try putting the code in ltac2_ltac1_plugin instead

Reply:
Is this the situation with other tactic plugins as well?  Should they all be split at once?

Reply:
Most plugins seem to only depend on Ltac1 for the mlg declaration, but in some cases it depends in a more intricate way, e.g. tauto is basically a Ltac1 library.

Reply:
I could delete the coq-core.plugins.ltac dependency in cc_plugin and have a folder somewhere for Ltac notation definitions. Maybe similar to Ltac2 in the user-contrib folder, or in plugins as ltac_notations. But if some plugins can't be split this way it may be unintuitive why some have seperate notation definitions and some not.

Reply:
IIUC this is the last point that is blocking this PR. Should we split the congruence plugin in half then?

Reply:
Is Ltac1 Tactic Notation good enough that if we add ltac2 bindings for congruence we could define ltac1 congruence that way?
Although even if it is, we would still need to figure out how to get prelude depending on ltac2